docs/DeveloperGuide.md

```diff
@@ -1,38 +1,726 @@
+  used
+  by all commands, a `QuestionList` object is passed by reference to construct command
+  objects(`LikeCommand`, `GetCommand`,
+  `HomeCommand`, `FavoriteCommand`). Hence, it is a dependency to all command objects in Self Reflection. This structure
+  allows
+  data to be centralised and well organised by one class.
+- By abstracting the above-mentioned attributes and methods as a separate class instead of putting them
+  in `ReflectionManager`,
+  the `ReflectionManager` class can solely focus command execution. All the data related to the list of questions is
+  stored
+  in `QuestionList` class. As such, Single responsibility can be better achieved.
+
+`ReflectUi` class: <br>
+
+- This subclass is created to allow Self Reflection feature to have more customised output behaviour(e.g. type of
+  separators)
+  other than those inherited from parent class `TextUi`.
+
+`RandomNumberGenerator` class; <br>
+
+- It is used generate a set of 5 distinct integers(0 ~ num_of_questions-1), this set
+  of integers will be used as indexes to select the corresponding questions from the pool of 10 questions available.
+
+#### Command Package
+
+`GetCommand` class: <br>
+
+- Command format: `get`
+- This command allows users to get a list of 5 random introspective questions.
+- The commands are validated by the `validateCommand()` method and a `BadCommandException` will be thrown if the
+  commands are invalid.
+- A `QuestionList` object is passed in as a dependency to generate the set of indexes and provide the pool of 10
+  introspective questions available.
+
+`LikeCommand` class: <br>
+
+- Command format: `like <index of question(1~5)>`
+- Users can add an introspective question that is generated in the previous set into their favorite list. Since there
+  will only
+  be 5 questions per set, the indexes are restricted to integer 1~5.
+- `addFavQuestion()` method in `QuestionList` class is used to add and store the data.
+- Users can only successfully add a question to favorite list if they have gotten a set of questions previously.
+- Every time a question is added into the favorite list, the indexes of this particular question will be stored in data
+  file straightaway. It prevents data loss due to unforeseen computer shutdown.
+
+`FavoriteCommand` class: <br>
+
+- Command format: `fav`
+- Users can get the questions in their favorite list.
+- `getFavQuestions()` method in `QuestionList` class matches the indexes to corresponding questions and return this set
+  of questions
+  back to `FavCommand` for output. As such, `QuestionList` is a dependency for `FavoriteCommand` as well.
+
+`HomeCommand` class: <br>
+
+- Command format: `home`
+- This command allows users to return back to the main WellNUS++ interface.
+- Similar to `GetCommand`, `validateCommand()` method will also be called to validate the command.
+- It will then call the class-level method `ReflectionManager.setIsExit()` to terminate the while loop
+  in `Reflectionmanager`.
```

Comment:
Perhaps it will also be great to include how these classes handle errors? Or if there are any cases (or zero cases) of errors they may need to handle?